### PR TITLE
[Do not merge] Add logging to ProgramNotificationSubscriberService to diagnose message loss

### PR DESCRIPTION
**Pull request by tivv**

### CODE REVIEW

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramNotificationSubscriberService.java [hunk 1]
```diff
@@ -294,12 +294,17 @@ protected void postProcess() {
       handleProgramEvent(programRunId, programRunStatus, notification, messageIdBytes,
                          appMetadataStore, programHeartbeatTable, result);
     }
-    if (clusterStatus == null) {
-      return result;
+    if (clusterStatus != null) {
+
```
Comment (sechegaray) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramNotificationSubscriberService.java:298]: nit: extra space.

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramNotificationSubscriberService.java [hunk 2]
```diff
@@ -294,12 +294,17 @@ protected void postProcess() {
...
+      handleClusterEvent(programRunId, clusterStatus, notification,
+                         messageIdBytes, appMetadataStore, context).ifPresent(result::add);
+    }
+
+    if (clusterStatus == null && programRunStatus == null) {
+        LOG.warn("Ignore notification without program or run status for program {}, {}",
```
Comment (sechegaray) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramNotificationSubscriberService.java:304]: nit: Change ignore for ignored

**Review by sechegaray — APPROVED** [review]
two minor nits

**Comment by tivv** [comment]
Closing this PR as it was mostly to get proper logs

**Pull request closed without merging**